packages/graphql-playground-react/src/components/Playground/ExecuteButtonOperation.tsx

```diff
@@ -8,31 +8,24 @@ export interface Props {
   highlight: any
 }
 
-export default class ExecuteButtonOperation extends React.Component<Props, {}> {
-  render() {
-    const { operation, highlight } = this.props
-    return (
-      <li
-        key={operation.name ? operation.name.value : '*'}
-        className={operation === highlight ? 'selected' : ''}
-        onMouseOver={this.handleMouseOver}
-        onMouseOut={this.handleMouseOut}
-        onMouseUp={this.handleMouseUp}
-      >
-        {operation.name ? operation.name.value : '<Unnamed>'}
-      </li>
-    )
-  }
+const ExecuteButtonOperation: React.SFC<Props> = ({
+  highlight,
+  onMouseOver,
+  onMouseOut,
+  onMouseUp,
+  operation,
+}) => (
+  <li
+    key={operation.name ? operation.name.value : '*'}
+    className={operation === highlight ? 'selected' : ''}
+    // tslint:disable-next-line
+    onMouseOver={() => onMouseOver(operation)}
```

Comment:
Using `React.SFC` sounds very interesting!
What I'm a little bit concerned about, however, is the use of arrow functions, which is an antipattern for normal components, as the function reference of the arrow function changes on each rerender, so do the props for the element. What do you think about that?

Reply:
I was actually blocked by tslint for this very reason and wasn't sure what it was talking about. I think this particular instance is fine because a majority of the props which could initiate a rerender are the ones that create the functions, which is why I added the tslint disable. I also learned my disable on tslint from another edited file which does the same thing for a simple lambda.

Long term my intention is to do another pass on these components and create a higher order component which passes a function directly into the SFC and only updates that specific function when components involved in it change. I was looking at recompose for this, but I'm open to suggestions.

For the short term, I'm not terribly concerned about the performance impact.

Reply:
Ok, thanks for giving your perspective on that one.
All other refactorings in this PR make a lot of sense to me, as they're simplifying a lot, but for this component, I don't see clear benefits for moving to this solution.

I'd rather stay with the normal component, which by the way could also be a pure react component.
We have [severe performance problems](https://github.com/graphcool/graphql-playground/issues/320) in the Playground (which I'm looking into right now) - we have to be really mindful of changes like these.

So my suggestion: Putting the arrow functions back to instance methods and turning the component into a `React.PureComponent`